scripts/ci-entrypoint.sh

```diff
@@ -146,6 +148,18 @@ get_cidrs() {
     fi
 }
 
+# get_cloud_provider determines if the Cluster is using an intree or external cloud-provider from the KubeadmConfigSpec.
+# any retryable operation in this function must return a non-zero exit code on failure so that we can
+# retry it using a `until get_cloud_provider; do sleep 5; done` pattern;
+# and any statement must be idempotent so that subsequent retry attempts can make forward progress.
+get_cloud_provider() {
+    CLOUD_PROVIDER=$("${KUBECTL}" get kubeadmcontrolplane -l cluster.x-k8s.io/cluster-name="${CLUSTER_NAME}" -o=jsonpath='{.items[0].spec.kubeadmConfigSpec.clusterConfiguration.controllerManager.extraArgs.cloud-provider}')
```

Comment:
I actually think it should be fine because we have the errexit option set in the ci-entrypoint.sh script

```
 cat test.sh
#!/bin/bash

set -o errexit
set -o nounset
set -o pipefail

get_cloud_provider() {
    STDOUT=$(ls dkfsadfsadf)
}

until get_cloud_provider; do
    sleep 5 
done
➜  cluster-api-provider-azure git:(oot-templates) ✗ ./test.sh  
ls: cannot access 'dkfsadfsadf': No such file or directory
ls: cannot access 'dkfsadfsadf': No such file or directory
ls: cannot access 'dkfsadfsadf': No such file or directory
ls: cannot access 'dkfsadfsadf': No such file or directory
^C
➜  cluster-api-provider-azure git:(oot-templates) ✗ echo $?
130
```

Reply:
ah! sgtm